### PR TITLE
netdata: update build config

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update build config and add support ipmi, dbengine, netfilter and cups.

cc @Mic92
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build netdata

File: pkgs/tools/system/netdata/default.nix [hunk 1]
```diff
@@ -10,14 +20,20 @@ stdenv.mkDerivation rec{
   };
 
   nativeBuildInputs = [ autoreconfHook pkgconfig ];
-  buildInputs = [ zlib ]
-    ++ (if stdenv.isDarwin then [ libossp_uuid CoreFoundation IOKit ] else [ libuuid ]);
+  buildInputs = [ curl.dev libcap.dev zlib.dev ]
```
Comment (Mic92) [inline pkgs/tools/system/netdata/default.nix]: `libcap` should be only included on non-macOS: https://logs.nix.ci/?key=nixos/nixpkgs.64241&attempt_id=bff9eca1-8826-4b1d-9d18-1a3c80d2a2be

**Comment by Izorkin** [comment]
@GrahamcOfBorg build netdata

**Comment by Mic92** [comment]
@GrahamcOfBorg build netdata

File: pkgs/tools/system/netdata/default.nix [hunk 2]
```diff
@@ -1,6 +1,16 @@
-{ stdenv, fetchurl, autoreconfHook, pkgconfig, zlib, libuuid, libossp_uuid, CoreFoundation, IOKit, lm_sensors }:
+{ stdenv, fetchurl, autoreconfHook, pkgconfig
+, CoreFoundation, IOKit, libossp_uuid
+, curl, libcap,  libuuid, lm_sensors, zlib
+, withCups ? false, cups
+, withDBengine ? true, libuv, lz4, judy
+, withIpmi ? true, freeipmi
```
Comment (Mic92) [inline pkgs/tools/system/netdata/default.nix]: put `!stdenv.isDarwin` here as well.

**Comment by Izorkin** [comment]
@GrahamcOfBorg build netdata

**Comment by Mic92** [comment]
@GrahamcOfBorg build netdata

**Comment by Izorkin** [comment]
@Mic92 disabled libmnl in darwin. Please check build.

**Comment by Izorkin** [comment]
@GrahamcOfBorg build netdata

**Comment by Mic92** [comment]
@GrahamcOfBorg build netdata

**Comment by Izorkin** [comment]
need replace `preConfigure = ''` to `preConfigure =  optionalString (!stdenv.isDarwin) ''` ?

**Comment by Mic92** [comment]
yes.

**Comment by Izorkin** [comment]
Update

**Comment by Izorkin** [comment]
@GrahamcOfBorg build netdata

**Comment by Izorkin** [comment]
@Mic92 how to add capabilities to freeipmi.plugin ?
`setcap cap_dac_override,cap_fowner+ep "/usr/libexec/netdata/plugins.d/freeipmi.plugin"`
https://github.com/netdata/netdata/issues/5534#issuecomment-469597852

**Comment by Mic92** [comment]
@GrahamcOfBorg build netdata

**Comment by Izorkin** [comment]
Thanks!

**Pull request merged**